tests/functional/adapter/empty/test_empty.py

```diff
@@ -7,3 +11,41 @@ class TestSnowflakeEmpty(BaseTestEmpty):
 
 class TestSnowflakeEmptyInlineSourceRef(BaseTestEmptyInlineSourceRef):
     pass
+
+
+class TestMetadataWithEmptyFlag:
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {"my_seed.csv": _models.SEED}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "schema.yml": _models.SCHEMA,
+            "control.sql": _models.CONTROL,
+            "get_columns_in_relation.sql": _models.GET_COLUMNS_IN_RELATION,
+            "alter_column_type.sql": _models.ALTER_COLUMN_TYPE,
+            "alter_relation_comment.sql": _models.ALTER_RELATION_COMMENT,
+            "alter_column_comment.sql": _models.ALTER_COLUMN_COMMENT,
+            "alter_relation_add_remove_columns.sql": _models.ALTER_RELATION_ADD_REMOVE_COLUMNS,
+            "truncate_relation.sql": _models.TRUNCATE_RELATION,
+        }
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup(self, project):
+        run_dbt(["seed"])
+
+    @pytest.mark.parametrize(
+        "model",
+        [
+            "control",
+            "get_columns_in_relation",
+            "alter_column_type",
+            "alter_relation_comment",
+            "alter_column_comment",
+            "alter_relation_add_remove_columns",
+            "truncate_relation",
+        ],
+    )
+    def test_run(self, project, model):
```

Comment:
Some of these tests pass even without fixing `{{ relation }}` to be `{{ relation.render() }}`. I think it has something to do with whether the template returns just the sql or whether it does the call. I added the tests in case this changes in the future, and updated the macros to use `{{ relation.render() }}`.